capitalone/rubicon-ml
# add failure modes

## What
  * adds failure modes to rubicon to log or warn on exceptions rather than raising them
    * adds `set_failure_mode` and the `failsafe` decorator to `rubicon_ml.client.utils.exception_handling`
    * decorates each public facing client method with `@failsafe`
  * adds example to docs
    * adds new tutorial section to docs' source
      * TODO (future): reorganize these source files to match the structure of the rendered docs

## How to Test
  * run through the new example, `notebooks/tutorials/failure-modes.ipynb`, and make sure it works as expected
  * build the docs locally and validate that the new example renders properly and the new `exception_handling` module is present in the API reference
